filament/backend/src/metal/MetalBuffer.h

```diff
@@ -57,18 +84,29 @@ class TrackedMetalBuffer {
     id<MTLBuffer> get() const noexcept { return mBuffer; }
     operator bool() const noexcept { return bool(mBuffer); }
 
-    static uint64_t getAliveBuffers() { return aliveBuffers; }
+    static uint64_t getAliveBuffers() {
+        uint64_t sum = 0;
+        for (const auto& v : aliveBuffers) {
+            sum += v;
+        }
+        return sum;
+    }
+
+    static uint64_t getAliveBuffers(Type type) {
+        assert_invariant(type != Type::NONE);
+        return aliveBuffers[toIndex(type)];
+    }
 
 private:
     void swap(TrackedMetalBuffer& other) noexcept {
-        id<MTLBuffer> temp = mBuffer;
-        mBuffer = other.mBuffer;
-        other.mBuffer = temp;
+        std::swap(mBuffer, other.mBuffer);
+        std::swap(mType, other.mType);
     }
 
     id<MTLBuffer> mBuffer;
+    Type mType = Type::NONE;
 
-    static std::atomic<uint64_t> aliveBuffers;
+    static std::array<uint64_t, TypeCount> aliveBuffers;
```

Comment:
should probably be `mAliveBuffers`

Reply:
Leaving off the `m` prefix since it's `static`